src/subtask.ts

```diff
@@ -5,6 +5,7 @@ export class ULabelSubtask {
     public actions: { stream: any[]; undone_stack: any[]; };
     public class_ids: number[] = [];
     public class_defs: ClassDefinition[]
+    public active_annotation: ULabelAnnotation
```

Comment:
i believe active_annotation is actually the annotation id, so a `string` 

tbh this is a property that is set by `suggest_edits()` and can probably just be consolidated with the existing `active_id` field

Reply:
I tried to consolidate with `.state.active_id` in b7ea252, but this badly breaks the dialogs (try to drag an annotation and you end up making a new one).

I could change this to `active_dialog_ann_id` to make it more verbose if we want to leave it for now. Otherwise I'll have to spend some time digging to see what else is interfering.

Reply:
Or `active_edit_dialog_ann_id` or something along those lines. Just needs to be differentiated from the other usage.

Reply:
maybe `state.edit_candidate.annid` ?

Reply:
I did a full breakdown of their usage, only looking at spots the values are explicitly set to something non-null.

Spots `active_id` is set:
- `begin_annotation`
- `start_complex_polygon`
- `begin_brush`
- `begin_edit`
- `begin_move`
- `handle_mouse_move` (which immediately calls `start_complex_polygon`)

Spots `active_annotation` is set:
- End of `suggest_edits`

Spots `edit_candidate` is set:
- `suggest_edits` (twice, depending on the annotation type)

I don't think we necessarily need to squash them, but their names should reflect the flow of logic (Hover > Edit Dialog > User Action > Actual Annotation Changes), and the transition between that logic should be clear. A single enum for global state might be worth investing the time to figure out.

Reply:
i think `edit_candidate` matches that flow, it is generated by `suggest_edits()` and then acted upon if the dialog is clicked

Reply:
OK I reverted b7ea252 and made the changes in c599376. Nothing appears broken.